### DISCUSSION TITLE
强制使段落缓存失效并重新调用翻译

### DISCUSSION THREAD

**Discussion by function2-llx** [immersive-translate/immersive-translate#2342] [Ideas]
### 新功能 | New Feature

如题，希望能提供强制使网页上某个段落的翻译缓存失效，并重新对其调用相应翻译接口的功能。

### 为什么您想要添加这些功能 | Why you want to add these features

有时候因为各种各样的问题（如翻译接口自身问题、网络原因、沉浸式翻译自身问题等），缓存中的结果可能并没有对应翻译接口正常应返回的结果，此时我会希望能使相应缓存失效并重新调用翻译接口。然而，目前沉浸式翻译中只能删除全部的缓存，这样对于收费接口不太经济。

此问题在 https://github.com/immersive-translate/immersive-translate/issues/760 中的讨论中有涉及到，但因时间较远故新开讨论。

### 提议 | Proposal

类似于鼠标悬停翻译段落的方式，实现相应强制重新翻译制定段落。

**Comment by theowenyoung** [comment]
似乎目前的行为里，鼠标悬停某一段就是强制会刷新缓存。 @keyboard3 是吗？

**Reply by function2-llx** [reply]
根据平时的使用经验，鼠标悬停已经被翻译过的段落，结果响应速度仍然很快，应该是在使用缓存。

**Reply by keyboard3** [reply]
相同翻译服务会复用缓存，可以指定其他翻译服务

**Reply by keyboard3** [reply]
插件不会缓存错误。除非接口返回的内容本身就是有误。而这种情况你短期重试也不会有帮助

**Reply by function2-llx** [reply]
我再举例举个例子说明下问题吧。我一直使用 DeepSeek 官方接口进行翻译，近期由于各种原因它的接口变得不太稳定，有时会出问题有时又恢复，然后估计也是因为这个原因导致返回结果有时会出错，并且也被插件缓存了，如下图：

![image](https://github.com/user-attachments/assets/231aa965-c34b-4db4-9a6c-6fb3c3ac8f53)

**Comment by Ricardo-Evans** [comment]
同样需要这个功能，希望鼠标悬停翻译能使该段缓存失效，在AI翻译场景下经常会需要此类重试

**Reply by function2-llx** [reply]
和悬停翻译还是要分开，因为有时候也会想手动开关一下部分段落的翻译
